### PR TITLE
Add an option to clamp HDR exposure to reduce environment fireflies

### LINKED ISSUE #886: Implement importing HDR images as tonemapped LDR (8-bit) images
**Describe the project you are working on:**
3d multiplayer game

**Describe the problem or limitation you are having in your project:**
I want to use openexr for standard texture usage too.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

Take a hdr (openexr) image, unable to make it a 8bit image.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

Take the hdr image and tone map it.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Image modification is cpu intensive.

**Is there a reason why this should be core and not an add-on in the asset library?:**

Import code is already being owned by the built in openexr module, so have to modify the openxrr module.

### PR DESCRIPTION
**Pull request by Calinou**
`master` version of https://github.com/godotengine/godot/pull/60588.

HDRI panoramas designed for realistic lighting can have extremely bright suns, causing fireflies to appear in the environment reflection (in addition to making environment lighting too bright when a DirectionalLight3D is used).

This uses the Filament tonemapping formula.

This closes https://github.com/godotengine/godot-proposals/issues/886.

**Testing project:** [test_radiance_map_high_quality_1.zip](https://github.com/godotengine/godot/files/8075791/test_radiance_map_high_quality_1.zip)

## Preview

*I recommend comparing non-clamped versions without a DirectionalLight3D against clamped versions that use a DirectionalLight3D.*

#### High Quality filter mode

| Without clamping | With clamping |
|-|-|
| 2022-02-18_18 50 28_720p | 2022-02-18_18 50 55_720p |

#### Real-Time filter mode

| Without clamping | With clamping |
|-|-|
| 2022-02-18_18 50 41_720p | 2022-02-18_18 51 07_720p |

#### High Quality filter mode

*Looks too bright without clamping, but brightness is good with clamping.*

| Without clamping | With clamping |
|-|-|
| 2022-02-18_18 51 28_720p | 2022-02-18_18 51 49_720p |

#### Real-Time filter mode

*Looks too bright without clamping, but brightness is good with clamping.*

| Without clamping | With clamping |
|-|-|
| 2022-02-18_18 51 37_720p | 2022-02-18_18 51 58_720p |

### CODE REVIEW

**Comment by fire** [comment]
I argue we should be enabled on but kept as a boolean setting for when we're able to render HDRIs with the proper tone mapping and rendering lux changes.

See also https://github.com/godotengine/godot-proposals/issues/723

**Comment by clayjohn** [comment]
We should use a tone-mapping clamp similar to Filament https://github.com/google/filament/blob/80f13a8149147ce733cadf896931af5f79ead73e/libs/ibl/src/CubemapUtils.cpp#L38-L57

Clamping to 4 is much too low. It essentially renders the HDRIs meaningless. Filament seems to have had luck clamping to 16384 which seems like a much more reasonable value (for reference the example HDRI you have above goes from 0-65505).

**Comment by Calinou** [comment]
Rebased to use Filament's tonemapping formula (thanks @clayjohn for the suggestion). I took new screenshots with radiance map prefiltering.

File: editor/import/resource_importer_texture.cpp [hunk 1]
```diff
@@ -411,6 +412,7 @@ Error ResourceImporterTexture::import(const String &p_source_file, const String
 	const bool stream = p_options["compress/streamed"];
 	const int size_limit = p_options["process/size_limit"];
 	const bool hdr_as_srgb = p_options["process/hdr_as_srgb"];
+	const float hdr_clamp_exposure = p_options["process/hdr_clamp_exposure"];
```
Comment (clayjohn) [inline editor/import/resource_importer_texture.cpp:415]: I guess this should be a bool?

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**